OpenSim/Moco/MocoGoal/MocoContactImpulseTrackingGoal.cpp

```diff
@@ -179,76 +175,84 @@ void MocoContactTrackingGoal::initializeOnModelImpl(const Model& model) const {
-    }
+        // If we didn't find a MocoScaleFactor for this force direction, set
+        // the reference pointer to null.
+        if (!foundScaleFactor) {
+            thisScaleFactorRef = nullptr;
+        }
+        m_scaleFactorRefs.push_back(thisScaleFactorRef);
+        
+        // Check that the length of scale factors == 1; As the goal does not
+        // support multiple integrals currently
+        if (m_scaleFactorRefs.size() != 1)
+            OPENSIM_THROW_FRMOBJ(Exception,
+                "Expected size of scale factors to be 1, but "
+                " got '{}.",
+                m_scaleFactorRefs.size());
 
-    // Should the contact force errors be projected onto a plane or vector?
-    if (get_projection() == "vector") {
-        m_projectionType = ProjectionType::Vector;
-    } else if (get_projection() == "plane") {
-        m_projectionType = ProjectionType::Plane;
-    } else if (get_projection() != "none") {
-        OPENSIM_THROW_FRMOBJ(Exception,
-                "Expected 'projection' to be 'none', 'vector', or 'plane', but "
-                "got '{}'.",
-                get_projection());
-    }
-    if (m_projectionType != ProjectionType::None) {
-        OPENSIM_THROW_IF_FRMOBJ(getProperty_projection_vector().empty(),
-                Exception, "Must provide a value for 'projection_vector'.");
-        m_projectionVector = SimTK::UnitVec3(get_projection_vector());
     }
 
+    // Which axis should the impulse be tracked for?
+    if (get_impulse_axis() < 0 || get_impulse_axis() > 2)
+        OPENSIM_THROW_FRMOBJ(Exception,
+            "Expected 'impulse_axis' to be either 0, 1 or 2, but "
+            "got '{}'.",
+            get_impulse_axis());
+
+    // Check the number of contact groups which have been added; 
+    // The MocoImpulseTrackingGoal currently only supports tracking 
+    // the impulse from a single limb at a time
+    if (getProperty_contact_groups().size() > 1)
+        OPENSIM_THROW_FRMOBJ(Exception,
+            "Expected a single (1) contact group, but "
+            "got '{}'.",
+            getProperty_contact_groups().size());
+    
     setRequirements(1, 1, SimTK::Stage::Velocity);
 }
 
-int MocoContactTrackingGoal::findRecordOffset(
-        const MocoContactTrackingGoalGroup& group,
-        const SmoothSphereHalfSpaceForce& contactForce,
-        const std::string& appliedToBody) const {
+int MocoContactImpulseTrackingGoal::findRecordOffset(
+    const MocoContactImpulseTrackingGoalGroup& group,
+    const SmoothSphereHalfSpaceForce& contactForce,
+    const std::string& appliedToBody) const {
 
     // Is the ExternalForce applied to the sphere's body?
     const auto& sphereBase =
-            contactForce.getConnectee<ContactSphere>("sphere")
-                    .getConnectee<PhysicalFrame>("frame")
-                    .findBaseFrame();
+        contactForce.getConnectee<ContactSphere>("sphere")
+        .getConnectee<PhysicalFrame>("frame")
+        .findBaseFrame();
```

Comment:
Similar to the if-statement expression, if you are chaining function calls like this, you should indent subsequent lines.